### PR TITLE
Bau check your answers

### PR DESCRIPTION
**Pull request by SteveSugden**
Fix for problem of displaying agent details section when trader
Unit Test for CheckYourAnswersHelper to improve code coverage

### CODE REVIEW

File: app/viewmodels/RepeaterAnswerSection.scala [hunk 1]
```diff
@@ -20,4 +20,6 @@ case class RepeaterAnswerSection (headingKey: String,
                                   relevanceRow: AnswerRow,
                                   rows: Seq[RepeaterAnswerRow],
                                   addLinkKey: String,
-                                  addLinkUrl: String) extends Section
+                                  addLinkUrl: String) extends Section {
+  override def isEmpty: Boolean = rows.isEmpty
```
Comment (gokyo) [inline app/viewmodels/RepeaterAnswerSection.scala]: is the file name (and the class name) correct?
`RepeaterAnswerSection`?
Did you mean `Repeated`?

Comment (SteveSugden) [reply app/viewmodels/RepeaterAnswerSection.scala]: I don't know.  It's an existing class.  Don't think we use it.

**Comment by gokyo** [comment]
`rows` are used by both children of `Section`
You could define it on the parent trait and define the method `isEmtpy` (with a concrete implementation) there, so you avoid repeating the same code

File: app/viewmodels/RepeaterAnswerSection.scala (hunk 1 above)
Comment (gokyo) [reply app/viewmodels/RepeaterAnswerSection.scala]: You could fix this typo

Comment (gokyo) [reply app/viewmodels/RepeaterAnswerSection.scala]: we could check what is the purpose of these classes and do some cleaning

File: test/unit/views/CheckYourAnswersViewSpec.scala [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+      text must include("Your details")
+      text must include("Trader Details")
+      text must not include("Details of the business, organisation or individual you represent")
+    }
+
+    "contain the answers for an agent" in {
+      val text = asDocument(createAgentView()).text()
+
+      text must include("Your details")
```
Comment (gokyo) [inline test/unit/views/CheckYourAnswersViewSpec.scala]: is `include` the best way to assert? I think we should assert by equality (exact match)

File: app/views/check_your_answers.scala.html [hunk 3]
```diff
@@ -31,7 +31,7 @@
 
         @components.heading(messages("checkYourAnswers.heading"))
 
-        @for(section <- answerSections){
+        @for(section <- answerSections if !section.isEmpty){
```
Comment (gokyo) [inline app/views/check_your_answers.scala.html]: you can make it more readable:
`if section.nonEmpty`

File: test/unit/views/CheckYourAnswersViewSpec.scala (hunk 2 above)
Comment (SteveSugden) [reply test/unit/views/CheckYourAnswersViewSpec.scala]: Exact match on a view?

Comment (gokyo) [reply test/unit/views/CheckYourAnswersViewSpec.scala]: sorry, my fault. 
I did not realise you are comparing the whole HTML content

**Pull request merged**